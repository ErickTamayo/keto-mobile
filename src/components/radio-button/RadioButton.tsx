import * as React from 'react'
import { TouchableOpacity, Text, View, StyleProp, ViewStyle } from 'react-native'
import {
  checkedOutlineStyle,
  uncheckedOutlineStyle,
  checkedFillStyle,
  uncheckedFillStyle,
  checkedTextStyle,
  uncheckedTextStyle,
  radioContainerStyle,
  containerStyle,
  wrapperStyle,
} from './styles'

export interface Props {
  title: string
  variant: 'lg'
  onPress?: (value: string) => void
  disabled?: boolean
  checked: boolean
  value: string
  style?: StyleProp<ViewStyle>
}

export default class RadioButton extends React.Component<Props, {}> {
  private handleOnPress = () => {
    const { onPress, value } = this.props
    onPress && onPress(value)
  }

  render(): JSX.Element {
    const { checked, disabled, title, style } = this.props
    return (
      <View style={[...containerStyle, style]}>
        <View style={wrapperStyle}>
          <View style={radioContainerStyle}>
            <TouchableOpacity
              style={[...(checked ? checkedOutlineStyle : uncheckedOutlineStyle)]}
              onPress={this.handleOnPress}
              disabled={disabled}
            >
              <View style={[...(checked ? checkedFillStyle : uncheckedFillStyle)]} />
            </TouchableOpacity>
          </View>
          <Text style={[...(checked ? checkedTextStyle : uncheckedTextStyle)]}>{title}</Text>
        </View>
      </View>
    )
  }
}
