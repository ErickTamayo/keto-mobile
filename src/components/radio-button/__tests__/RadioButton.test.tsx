import 'react-native'
import React from 'react'
import { TouchableOpacity, GestureResponderEvent } from 'react-native'
import { shallow } from 'enzyme'
import RadioButton, { Props } from '../RadioButton'

describe('<RadioButton />', () => {
  it('renders correctly', () => {
    const props: Props = {
      title: 'radio-button-title',
      variant: 'lg',
      checked: true,
      value: 'radio-button-value',
    }

    const wrapper = shallow(<RadioButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders unchecked', () => {
    const props: Props = {
      title: 'radio-button-title',
      variant: 'lg',
      checked: false,
      value: 'radio-button-value',
    }

    const wrapper = shallow(<RadioButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders disabled', () => {
    const props: Props = {
      title: 'radio-button-title',
      variant: 'lg',
      checked: false,
      value: 'radio-button-value',
      disabled: true,
    }

    const wrapper = shallow(<RadioButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('triggers onPress once pressed', () => {
    const props: Props = {
      title: 'radio-button-title',
      variant: 'lg',
      checked: false,
      value: 'radio-button-value',
      onPress: jest.fn(),
    }

    const wrapper = shallow(<RadioButton {...props} />)
    wrapper.find(TouchableOpacity).prop('onPress')!({} as GestureResponderEvent)
    expect(props.onPress).toBeCalled()
  })
})
