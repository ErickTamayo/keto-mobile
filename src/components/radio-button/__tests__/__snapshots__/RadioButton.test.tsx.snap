// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "height": 60,
      },
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "column",
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "height": 36,
          },
          Object {
            "width": 36,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "borderRadius": 100,
            },
            Object {
              "borderWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "borderColor": "#57ae5b",
            },
          ]
        }
      >
        <Component
          style={
            Array [
              Object {
                "borderRadius": 100,
              },
              Object {
                "height": 29.5,
                "width": 29.5,
              },
              Object {
                "backgroundColor": "#57ae5b",
              },
            ]
          }
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "fontSize": 12,
          },
          Object {
            "marginTop": 13,
          },
          Object {
            "color": "#57ae5b",
          },
        ]
      }
    >
      radio-button-title
    </Component>
  </Component>
</Component>
`;

exports[`<RadioButton /> renders disabled 1`] = `
<Component
  style={
    Array [
      Object {
        "height": 60,
      },
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "column",
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "height": 36,
          },
          Object {
            "width": 36,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        disabled={true}
        onPress={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "borderRadius": 100,
            },
            Object {
              "borderWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "borderColor": "#cfcfcf",
            },
          ]
        }
      >
        <Component
          style={
            Array [
              Object {
                "borderRadius": 100,
              },
              Object {
                "height": 29.5,
                "width": 29.5,
              },
            ]
          }
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "fontSize": 12,
          },
          Object {
            "marginTop": 13,
          },
          Object {
            "color": "#cfcfcf",
          },
        ]
      }
    >
      radio-button-title
    </Component>
  </Component>
</Component>
`;

exports[`<RadioButton /> renders unchecked 1`] = `
<Component
  style={
    Array [
      Object {
        "height": 60,
      },
      Object {
        "alignItems": "center",
      },
      undefined,
    ]
  }
>
  <Component
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "flexDirection": "column",
        },
        Object {
          "alignItems": "center",
        },
      ]
    }
  >
    <Component
      style={
        Array [
          Object {
            "height": 36,
          },
          Object {
            "width": 36,
          },
        ]
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "borderRadius": 100,
            },
            Object {
              "borderWidth": 1,
            },
            Object {
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
            },
            Object {
              "borderColor": "#cfcfcf",
            },
          ]
        }
      >
        <Component
          style={
            Array [
              Object {
                "borderRadius": 100,
              },
              Object {
                "height": 29.5,
                "width": 29.5,
              },
            ]
          }
        />
      </TouchableOpacity>
    </Component>
    <Component
      style={
        Array [
          Object {
            "fontFamily": "avenir-next-medium",
          },
          Object {
            "fontSize": 12,
          },
          Object {
            "marginTop": 13,
          },
          Object {
            "color": "#cfcfcf",
          },
        ]
      }
    >
      radio-button-title
    </Component>
  </Component>
</Component>
`;
